src/harness/vfsUtil.ts

```diff
@@ -167,16 +167,15 @@ namespace vfs {
          *
          * @link http://pubs.opengroup.org/onlinepubs/9699919799/functions/time.html
          */
-        public time(value?: number | Date | (() => number | Date)): number {
-            if (value !== undefined && this.isReadonly) throw createIOError("EPERM");
-            let result = this._time;
-            if (typeof result === "function") result = result();
-            if (typeof result === "object") result = result.getTime();
-            if (result === -1) result = Date.now();
+        public time(value?: number): number {

```

Comment:
I'm not sure I agree with this change. This is something that `time` already supports by passing a callback to the `FileSystem` constructor. You could implement what you need like this:

```ts
let time = 0;
const fs = new vfs.FileSystem(/*ignoreCase*/ true, {
  time: () => time += ts.TestFSWithWatch.timeIncrements
});
```

I do agree that freezing the time when an FS becomes readonly makes sense (i.e., changing `makeReadOnly` to do `this._time = this.time()` before freezing).

Reply:
time function was used for tsc -- build tests so i dont see why we cant simplify it

Reply:
I added it as a general purpose mechanism to be able to control the reported time in unit tests, since it seemed like something we would like to mock/fake when necessary. Most of the tests in compiler, conformance, fourslash, etc. don't depend on the time. I suppose we can drop this capability if the only things that use it expect the auto-increment behavior.